src/Core/App.xaml.cs

```diff
@@ -85,10 +85,27 @@ public static ResumeWindow CurrentWindow
             }
         }
 
-        //Allows setting Options from MainActivity before base.OnCreate
-        public void SetOptions(AppOptions appOptions)
+        /// <summary>
+        /// Allows setting Options from MainActivity before base.OnCreate
+        /// Note 1: This is only be used by Android due to way it's Lifecycle works
+        /// Note 2: This method does not replace existing Options in App.xaml.cs if it exists already.
+        ///         It only updates properties in Options related with Autofill/CreateSend/etc..
+        /// </summary>
+        /// <param name="appOptions">Options created in Android MainActivity.cs</param>
+        public void SetAndroidOptions(AppOptions appOptions)
         {
-            Options = appOptions ?? new AppOptions();
+            if (Options == null)
+            {
+                Options = appOptions ?? new AppOptions();
+            }
+            else if(appOptions != null)
+            { 
+                Options.Uri = appOptions.Uri;
+                Options.MyVaultTile = appOptions.MyVaultTile;
+                Options.GeneratorTile = appOptions.GeneratorTile;
+                Options.FromAutofillFramework = appOptions.FromAutofillFramework;
+                Options.CreateSend = appOptions.CreateSend;
+            }
         }
```

Comment:
⛏️ If this method is only for Android then I'd wrap it in `#if ANDROID` so this method is not added on iOS where it's not going to be used and also if for some reason someone calls it from iOS with the precompiled directive it will throw an error on compilation.
🤔 Additionally, @mpbw2 can't remember  but is there a reason why we are manually assigning the properties instead of just using `void SetAllFrom(AppOptions o)`? Like `Options.SetAllFrom(appOptions)`; are we afraid some property will be overridden with a wrong value?

Reply:
@fedemkr That's my impression based on how the original (Xamarin) flow worked.  It was a safety measure to prevent mixing of Options "states".  Is that correct @dinisvieira ?

Reply:
@fedemkr @mpbw2 I did it differently only because I completely missed the `SetAllFrom()` method. 😄 
That said, it would still result in a bug in MAUI because it's going to replace the `HideAccountSwitcher` property (possibly others?).
I could still improve this by adding a new `SetAutofillOptions()` method with this code inside the `AppOptions` class wrapped in a` #if ANDROID`
What do you think?

Reply:
@fedemkr @mpbw2 Oh wait, I just remembered that it can be null in some initialization scenarios.
That's likely why I was using the `SetOptions` directly on `App.xaml.cs`.
It's also already wrapped in a `#if ANDROID`
Should I just rename it back to `SetOptions`? (instead of `SetAndroidOptions`)

Reply:
> Should I just rename it back to `SetOptions`?

The more I look at it the more I think yes, `SetOptions` is better

Reply:
Oh missed the outer `#if ANDROID` sorry. Thanks for the info! It's ok as it is, let's leave it like this to avoid introducing secondary effects.

Reply:
About naming, whatever you feel is better is fine by me, it's only going to be used on Android so both options makes good sense to me.

Reply:
@dinisvieira Just leave it as is 👍